source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -90,59 +90,86 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
         return SetStyleFromResourceDictionary(renderContext, HStringReference(resourceName).Get(), frameworkElement);
     }
 
+    void SetStyleFromResourceDictionary(winrt::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderContext renderContext,
+                                        winrt::hstring resourceName,
+                                        winrt::Windows::UI::Xaml::IFrameworkElement frameworkElement)
+    {
+        THROW_IF_FAILED(SetStyleFromResourceDictionary(
+            renderContext.as<ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveRenderContext>().get(),
+            std::wstring_view(resourceName).data(),
+            frameworkElement.as<ABI::Windows::UI::Xaml::IFrameworkElement>().get()));
+    }
+
     HRESULT XamlHelpers::SetSeparatorVisibility(_In_ IPanel* parentPanel)
     {
         // Iterate over the elements in a container and ensure that the correct separators are marked as visible
         ComPtr<IVector<UIElement*>> children;
         RETURN_IF_FAILED(parentPanel->get_Children(&children));
 
         bool foundPreviousVisibleElement = false;
-        IterateOverVector<UIElement, IUIElement>(children.Get(), [&](IUIElement* child) {
-            ComPtr<IUIElement> localChild(child);
-
-            ComPtr<IFrameworkElement> childAsFrameworkElement;
-            RETURN_IF_FAILED(localChild.As(&childAsFrameworkElement));
-
-            // Get the tag for the element. The separators themselves will not have tags.
-            ComPtr<IInspectable> tag;
-            RETURN_IF_FAILED(childAsFrameworkElement->get_Tag(&tag));
-
-            if (tag)
-            {
-                ComPtr<IElementTagContent> elementTagContent;
-                RETURN_IF_FAILED(tag.As(&elementTagContent));
-
-                ComPtr<IUIElement> separator;
-                RETURN_IF_FAILED(elementTagContent->get_Separator(&separator));
-
-                Visibility visibility;
-                RETURN_IF_FAILED(child->get_Visibility(&visibility));
+        IterateOverVector<UIElement, IUIElement>(children.Get(),
```

Comment:
probably work for a different PR, but I'm excited about the possibility of using the cppwinrt [collections stuff](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/collections) instead of our bespoke iterating thing :)

Reply:
That looks nifty. I will take a look as I move forward.